app/contact/components/contact-form.tsx

```diff
@@ -0,0 +1,259 @@
+    Form,
+    FormControl,
+    FormField,
+    FormItem,
+    FormLabel,
+    FormMessage,
+} from "@/components/ui/form";
+import { toast } from "sonner";
+import { sendContactEmail } from "../actions";
+import { SendEmailRequest } from "@/lib/fusion/workshop/email.service.pb";
+
+const interestsSchema = z.object({
+    workshops: z.boolean().optional(),
+    woodwork: z.boolean().optional(),
+    metalwork: z.boolean().optional(),
+    cnc: z.boolean().optional(),
+    engineering: z.boolean().optional(),
+    memberships: z.boolean().optional(),
+});
+
+const formSchema = z.object({
+    firstName: z.string().min(1, "First name is required"),
+    lastName: z.string().min(1, "Last name is required"),
+    email: z.string().email("Invalid email address"),
+    contactNumber: z.string().min(1, "Contact number is required"),
+    interests: interestsSchema,
+    message: z.string().min(1, "Message is required"),
+    newsletter: z.boolean().optional(),
+});
+
+type ContactFormValues = z.infer<typeof formSchema>
+
+export const ContactForm = () => {
+    const [emailSent, setEmailSent] = useState(false)
+    const form = useForm<ContactFormValues>({
+        resolver: zodResolver(formSchema),
+        defaultValues: {
+            firstName: "",
+            lastName: "",
+            email: "",
+            contactNumber: "",
+            interests: {
+                workshops: false,
+                woodwork: false,
+                metalwork: false,
+                cnc: false,
+                engineering: false,
+                memberships: false,
+            },
+            message: "",
+            newsletter: false,
+        },
+    });
+
+    const onSubmit = async (values: z.infer<typeof formSchema>) => {
+        // Extract interests
+        const interests = Object.entries(values.interests)
+            .filter(([_, value]) => value)
+            .map(([key]) => key)
+            .join(", ");
+
+        // Build the email body
+
+        const plainTextContent = `
+            Name: ${values.firstName} ${values.lastName}
+            Email: ${values.email}
+            Contact Number: ${values.contactNumber}
+            Interests: ${interests}
+            Message: ${values.message}
+            Newsletter Subscription: ${values.newsletter ? 'Yes' : 'No'}
+        `;
+
+        const htmlContent = `
+            <div style="font-family: Arial, sans-serif; line-height: 1.6;">
+                <h2>Contact Form Submission</h2>
+                <p><strong>Name:</strong> ${values.firstName} ${values.lastName}</p>
+                <p><strong>Contact Number:</strong> ${values.contactNumber}</p>
+                <p><strong>Interests:</strong> ${interests}</p>
+                <p><strong>Message:</strong></p>
+                <p>${values.message}</p>
+                <p><strong>Newsletter Subscription:</strong> ${values.newsletter ? 'Yes' : 'No'}</p>
+            </div>
+        `;
+
+        const request: SendEmailRequest = {
```

Comment:
This should rather call a contact service's Submit method, sending the html etc from the client side is dangerous

Reply:
This logic can just be applied on the backend